lib/pure/ospaths.nim

```diff
@@ -423,12 +423,22 @@ proc isAbsolute*(path: string): bool {.rtl, noSideEffect, extern: "nos$1".} =
   ## Checks whether a given `path` is absolute.
   ##
   ## On Windows, network paths are considered absolute too.
+  runnableExamples:
+    doAssert "".isAbsolute.not
+    doAssert nil.isAbsolute.not
+    doAssert ".".isAbsolute.not
+    when defined(posix):
+      doAssert "/".isAbsolute
+      doAssert "a/".isAbsolute.not
+
+  if len(path) == 0: return false
+
   when doslikeFileSystem:
     var len = len(path)
-    result = (len > 0 and path[0] in {'/', '\\'}) or
```

Comment:
I fail to see how removing bounds checking helps with anything.

Reply:
the case `len=0` is already handled at the very top-level of this function